### PR TITLE
chore: adds example code and README for setting up testing with a payload project

### PR DESCRIPTION
**Pull request by JarrodMFlesch**
## Description

This adds a new `testing` folder inside the examples directory. The purpose is to show users how to get up and running with testing within their own payload projects.

- [x] I have read and understand the [CONTRIBUTING.md](../CONTRIBUTING.md) document in this repository.

## Type of change

- [x] Change to the [examples](../examples/) directory (does not affect core functionality)

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -2,8 +2,8 @@ module.exports = {
   verbose: true,
   testEnvironment: 'node',
   testMatch: [
-    '**/src/**/*.spec.ts',
-    '**/test/**/*int.spec.ts',
```
Comment (JarrodMFlesch) [inline jest.config.js:6]: Is anyone opposed to this change? It behaves the same as before, I just wanted to ask as I cannot see why we would need wildcards before the src and test entry points.

File: jest.config.js [hunk 2]
```diff
@@ -2,8 +2,8 @@ module.exports = {
...
+    '<rootDir>/src/**/*.spec.ts',
+    '<rootDir>/test/**/*int.spec.ts',
```
Comment (JarrodMFlesch) [inline jest.config.js:6]: Is anyone opposed to this change? It behaves the same as before, I just wanted to ask as I cannot see why we would need wildcards before the src and test entry points.

**Comment by jacobsfletch** [comment]
@JarrodMFlesch is this PR still valid? I see the testing example _did_ make its way into main. Do we need to close this one out?

**Comment by JarrodMFlesch** [comment]
@jacobsfletch I am going to merge this one into the 1.x branch. I see I need to adjust the testing example on 2.0 as well now too.

**Comment by jackadair** [comment]
Hey @JarrodMFlesch is the current example on 2.0? Just trying to figure out how to pass a different db url for testing but am getting the type error:
`Object literal may only specify known properties, and 'mongoURL' does not exist in type 'InitOptions'.`

Thanks!

**Comment by jackadair** [comment]
@JarrodMFlesch That would be amazing, thank you 🙏

**Pull request merged**